unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/GLTFast/Wrappers/GLTFTextureDownloaderWrapper.cs

```diff
@@ -15,34 +15,62 @@ public GltfTextureDownloaderWrapper(IWebRequestAsyncOperation asyncOp)
         }
 
         public bool Success => asyncOp.isSucceeded;
-        public string Error => asyncOp.webRequest.error;
-        public byte[] Data => asyncOp.webRequest.downloadHandler.data;
-        public string Text => asyncOp.webRequest.downloadHandler.text;
+        public string Error => asyncOp.webRequest?.error;
+        public byte[] Data => asyncOp.webRequest?.downloadHandler?.data;
+        public string Text => asyncOp.webRequest?.downloadHandler?.text;
         public bool? IsBinary => true;
 
         public bool MoveNext() =>
             asyncOp.MoveNext();
 
-        public Texture2D GetTexture(bool linear)
+        public Texture2D GetTexture(bool forceSampleLinear)
         {
-            Texture2D texture2D = new Texture2D(1, 1, TextureFormat.RGBA32, 0, linear);
+            Texture2D texture2D = new Texture2D(1, 1, TextureFormat.RGBA32, 0, forceSampleLinear);
 
-            if (asyncOp.webRequest.downloadHandler.data != null)
+            if (LoadTexture(texture2D))
+            {
+                texture2D = OptimizeTexture(forceSampleLinear, texture2D);
+                return texture2D;
+            }
+            else
+            {
+                Debug.Log("Failed to load texture with downloaded data");
+                DisposeTexture(texture2D);
+                return null;
+            }
+        }
+
+        private static void DisposeTexture(Texture2D texture2D)
+        {
+            if (Application.isPlaying)
```

Comment:
not sure if this can cause call for DestroyImmediate when application is paused. Maybe use `!Application.isEditor` instead?